src/sql/arrow_sql_gen/statement.rs

```diff
@@ -88,6 +92,12 @@ impl CreateTableBuilder {
     #[must_use]
     pub fn build_sqlite(self) -> String {
         self.build(SqliteQueryBuilder, &|f: &Arc<Field>| -> ColumnType {
+            // Sqlite does not natively support Arrays, Structs, etc
+            // so we use JSON column type for List, FixedSizeList, LargeList, Struct, etc
+            if f.data_type().is_nested() {
```

Comment:
Arrow map and union types are also nested, we can make a note inside the comments here, so when later on union / map is supported for sqlite, it's easier to locate this relevant changes